### PR TITLE
Add new processing approach for big S3 Objects (IOW-561)

### PR DESCRIPTION
**Pull request by kkehl-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Add new processing approach for big S3 objects
-----------
https://internal.cida.usgs.gov/jira/browse/IOW-561

Description
-----------
Add handling for big S3 objects using aws_s3 extension.  For now, keep switching on object size.  Once everything is working end to end, remove the if/else logic and process all S3 objects using the new approach.

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

**Review by ssoper-usgs — APPROVED** [review]

File: serverless.yml [hunk 1]
```diff
@@ -37,6 +41,8 @@ functions:
       DB_NAME: ${self:custom.db.connectInfo.DATABASE_NAME}
       DB_PASSWORD: ${self:custom.db.connectInfo.SCHEMA_OWNER_PASSWORD}
       AWS_DEPLOYMENT_REGION: ${self:provider.region}
+      BUCKET_NAME: iow-retriever-capture-${self:custom.environments.${self:provider.stage}}
+      S3_OBJECT_SIZE_LIMIT: 150000000
```
Comment (ayan-usgs) [inline serverless.yml:44]: It's actually the trigger that currently has the limit control: https://github.com/usgs/aqts-capture-trigger/blob/master/serverless.yml#L49.

File: serverless.yml [hunk 2]
```diff
@@ -37,6 +41,8 @@ functions:
       DB_NAME: ${self:custom.db.connectInfo.DATABASE_NAME}
       DB_PASSWORD: ${self:custom.db.connectInfo.SCHEMA_OWNER_PASSWORD}
       AWS_DEPLOYMENT_REGION: ${self:provider.region}
+      BUCKET_NAME: iow-retriever-capture-${self:custom.environments.${self:provider.stage}}
```
Comment (ayan-usgs) [inline serverless.yml]: I'm don't think this is necessary. The bucket name already comes in as part of the event that this lambda is invoked with: https://github.com/usgs/aqts-retriever-capture-raw-load/blob/master/src/etl/event_processor.py#L40

File: src/etl/rds.py [hunk 3]
```diff
@@ -84,6 +85,30 @@ def _execute_sql(self, sql, params=()):
...
+        aws_s3.table_import_from_s3(
+            'capture.json_data',
+            'start_time,response_time,response_code,url,api,script_name,script_pid,parameters,json_content,uuid',
+            'DELIMITER ''|''',
+            aws_commons.create_s3_uri(%s, %s,'us-west-2')
+        );"""
+
+        self._execute_sql(
+            insert_json_data, (
+                os.getenv('BUCKET_NAME'), object_key
```
Comment (ayan-usgs) [inline src/etl/rds.py]: See comment above.

File: src/load.py [hunk 4]
```diff
@@ -38,7 +49,13 @@ def lambda_handler(event, context):
     response = None
     try:
         logger.debug(event)
-        record_id = etl(event)
+        size = event['s3']['object']['size']
+        if size < os.getenv('S3_OBJECT_SIZE_LIMIT'):
```
Comment (ayan-usgs) [inline src/load.py:53]: The plan is to remove this `if` clause in the near future right (that's what's I have in mind anyway)?

**Review by ayan-usgs — CHANGES_REQUESTED** [review]
I think this will be good to go after the bucket name environment variable is removed.

File: serverless.yml (hunk 1 above)
Comment (ayan-usgs) [reply serverless.yml:44]: Unless it's changed in the trigger, those large files won't make it to this lambda.

File: src/load.py (hunk 4 above)
Comment (kkehl-usgs) [reply src/load.py:53]: correct

File: serverless.yml (hunk 2 above)
Comment (kkehl-usgs) [reply serverless.yml]: done.

**Comment by kkehl-usgs** [comment]
I made Andrew's requested changes and as per slack conversation he says this PR is good to go.

**Pull request merged**